server/src/test/java/io/crate/planner/optimizer/symbol/OptimizerTest.java

```diff
@@ -43,4 +45,24 @@ public void test_like_on_numeric_columns_keeps_cast_around_reference() throws Ex
             SymbolMatchers.isLiteral("10")
         ));
     }
+
+    @Test
+    public void test_comparison_of_string_and_numeric_doesnt_optimize_away_cast() throws Exception {
+        SQLExecutor e = SQLExecutor.builder(clusterService)
+            .addTable("create table tbl (strCol string, intCol int)")
+            .build();
+
+        //checking either sides of num <-> str comparison, reference casting remains
+        Symbol symbol1 = Optimizer.optimizeCasts(e.asSymbol("strCol::bigint > 3"), e.getPlannerContext(clusterService.state()));
+        Symbol symbol2 = Optimizer.optimizeCasts(e.asSymbol("intCol::string > '3'"), e.getPlannerContext(clusterService.state()));
```

Comment:
is there a difference for `::string` and `::bigint` casts or why do we have to test both?

Reply:
no difference, artefact from the previous implementation where I played around with (num,str)/(str,num) and wanted to test explicitly. will remove second test, for this implementation is redundant.